### PR TITLE
Updates play-java-intro to use PersonController and Hibernate 5.1.0.Final.

### PR DESCRIPTION
**Pull request by wsargent**
Updates `play-java-intro`:

* Updated from 4.3.7.Final to 5.1.0.Final
* Changed Person model to use Integer (was running into "Unknown integral data type for ids : java.lang.String")
* Changed Application.java to HomeController.java
* No longer use static `JPA.em()` methods, instead use `jpaAPI` as DI into the controller

### CODE REVIEW

File: templates/play-java-intro/app/controllers/HomeController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import views.html.index;
+
+import javax.inject.Inject;
+import java.util.List;
+
+import static play.libs.Json.toJson;
+
+public class HomeController extends Controller {
+
+    @Inject
```
Comment (marcospereira) [inline templates/play-java-intro/app/controllers/HomeController.java]: Shouldn't use constructor injection?

Comment (gmethvin) [inline templates/play-java-intro/app/controllers/HomeController.java]: @wsargent I agree with @marcospereira . All examples should use constructor injection. I think we eventually want to switch our docs to use it too, but at least any code that people base their apps on should do things the "right" way.

File: templates/play-java-intro/app/controllers/HomeController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import views.html.index;
+
+import javax.inject.Inject;
+import java.util.List;
+
+import static play.libs.Json.toJson;
+
+public class HomeController extends Controller {
+
+    private FormFactory formFactory;
```
Comment (benmccann) [inline templates/play-java-intro/app/controllers/HomeController.java]: I would make these fields `final`

**Comment by wsargent** [comment]
Should be good for review / merge now.

**Pull request merged**